product/domain/productBasics.go

```diff
@@ -570,6 +580,7 @@ func findMediaInProduct(p BasicProduct, group string, usage string) *Media {
 
 // IsInStock returns information if current product whether in stock or not
 func (bpd BasicProductData) IsInStock() bool {
+	// todo: not sure what to do here
```

Comment:
imho we should check the new field first and only if not set fall back to the deprecated field.
The logic would be: if any element of the stock array has InStock: true

In Addition we can add `IsInStockForDeliveryCode(deliveryCode string) bool`